server/com.dexels.navajo.listeners.continuations/src/com/dexels/navajo/server/listener/http/continuation/TmlRunnableBuilder.java

```diff
@@ -14,44 +14,44 @@
 
 public class TmlRunnableBuilder {
 
-
-	private TmlRunnableBuilder() {
-		// no instances
-	}
-	// Made static to indicate independence of fields
-	public static TmlRunnable prepareRunnable(
-			final HttpServletRequest req, HttpServletResponse resp, LocalClient localClient, String instance,long timeout)
-			throws IOException {
-		TmlContinuationRunner tmlRunner = (TmlContinuationRunner) req.getAttribute("tmlRunner");
-		if (tmlRunner != null) {
-			return null;
-		}
-
-		AsyncRequest request = constructRequest(req, resp, instance);
-		TmlContinuationRunner instantiateRunnable = new TmlContinuationRunner(request,localClient,timeout);
-		req.setAttribute("tmlRunner", instantiateRunnable);
-	    instantiateRunnable.setAttribute("tester", req.getHeader("X-Navajo-Tester") != null);
-		instantiateRunnable.suspendContinuation(resp);
-		return instantiateRunnable;
-	}
-
-
-	// Made static to indicate independence of fields
-	private static AsyncRequest constructRequest(final HttpServletRequest req,
-			HttpServletResponse resp, String instance)
-			throws IOException {
-		Object certObject = req.getAttribute("javax.servlet.request.X509Certificate");
-		String contentEncoding = req.getHeader("Content-Encoding");
-		String acceptEncoding = req.getHeader("Accept-Encoding");
-		AsyncRequest request = null;
-		if("POST".equals( req.getMethod())) {
-			request = new BaseRequestImpl(req, resp,acceptEncoding, contentEncoding, certObject,instance);
-		} else {
-			Navajo in = TmlHttpServlet.constructFromRequest(req);
-			request = new BaseRequestImpl(in, req,resp,instance);
-		}
-
-		return request;
+    public static TmlRunnable prepareRunnable(final HttpServletRequest req,
+            HttpServletResponse resp, LocalClient localClient, String instance, long timeout)
+            throws IOException {
+
+        TmlContinuationRunner tmlRunner = (TmlContinuationRunner) req.getAttribute("tmlRunner");
+        if (tmlRunner != null) {
+            return null;
```

Comment:
Some comment on this construction seems reasonable. I  suspect that it is done to prevent service request duplicates? If for some reason the network layer presents duplicate requests to the server??

Reply:
That is what I presumed as well. I did find I odd test. However, since it was outside the scope of my change, I left it alone.